src/Notifications/KillMail/AbstractKillMailNotification.php

```diff
@@ -0,0 +1,197 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace Herpaderpaldent\Seat\SeatNotifications\Notifications\KillMail;
+
+use Herpaderpaldent\Seat\SeatNotifications\Notifications\AbstractNotification;
+use Seat\Eveapi\Models\Killmails\KillmailDetail;
+use Seat\Eveapi\Models\Killmails\KillmailVictimItem;
+use Seat\Eveapi\Models\Market\Price;
+
+abstract class AbstractKillMailNotification extends AbstractNotification
+{
+    /**
+     * @var string
+     */
+    public $killmail_detail;
+
+    /**
+     * @var string
+     */
+    public $image;
+
+    /**
+     * AbstractKillMailNotification constructor.
+     */
+    public function __construct(int $killmail_id)
+    {
+
+        parent::__construct();
+
+        $this->killmail_detail = KillmailDetail::find($killmail_id);
+        $this->image = sprintf('https://imageserver.eveonline.com/Type/%d_64.png',
+            $this->killmail_detail->victims->ship_type_id);
+
+        array_push($this->tags, 'killmail_id:' . $killmail_id);
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getTitle(): string
+    {
+
+        return 'Kill Mail Notification';
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getDescription(): string
+    {
+
+        return 'Receive a notification about new kill mails.';
+    }
+
+    /**
+     * @return bool
+     */
+    final public static function isPublic(): bool
+    {
+
+        return true;
+    }
+
+    /**
+     * @return bool
+     */
+    final public static function isPersonal(): bool
+    {
+
+        return false;
+    }
+
+    /**
+     * @return array
+     */
+    final public static function getFilters(): ?string
+    {
+
+        return 'corporations';
+    }
+
+    /**
+     * @param $notifiable
+     *
+     * @return mixed
+     */
+    abstract public function via($notifiable);
+
+    public function resolveID($id, $is_alliance = false)
+    {
+
+        $cached_entry = cache('name_id:' . $id);
+
+        if (! is_null($cached_entry))
+            return $cached_entry;
+
+        if ($is_alliance)
+            return $this->getAllianceTicker($id);
+
+        // Resolve the Esi client library from the IoC
+        $eseye = app('esi-client')->get();
```

Comment:
Need a better way to do this and avoid to spam esi for every single sent notification.

Reply:
Ups forgot the cache ....